mne_bids/path.py

```diff
@@ -762,31 +840,47 @@ def _find_best_candidates(params, candidate_list):
     return best_candidates
 
 
-def _get_bids_fname_from_filesystem(*, bids_basename, bids_root, sub, ses,
-                                    kind):
+def _get_bids_fpath_from_filesystem(*, bids_basename, bids_root, sub, ses,
```

Comment:
is bids_basename here str or instance of bidspath? how do we know when is it which?

Reply:
We should probably discuss making use of Python's [type annotations](https://docs.python.org/3/library/typing.html) at one point.

Reply:
I think we would want something along these lines?

Do a refactor of `make_bids_basename` to `make_bids_fpath`, where then `bids_basename = bids_fpath.basename`, and `bids_fpath` is a BIDSPath object.

Reply:
The answer here is that it is just the str instance

Reply:
Yeah that makes sense. We should probably always use "path" somewhere in the variable name if it's a path object. 

I don't know about type annotations. An example would be nice, just for my curiosity ;-)